ext/date/php_date.c

```diff
@@ -3427,6 +3427,12 @@ static int timezone_initialize(php_timezone_obj *tzobj, const char *tz, size_t t
 	}
 
 	dummy_t->z = timelib_parse_zone(&tz, &dst, dummy_t, &not_found, DATE_TIMEZONEDB, php_date_parse_tzfile_wrapper);
+	if (!not_found && (*tz != '\0')) {
+		php_error_docref(NULL, E_WARNING, "Unknown or bad timezone (%s)", orig_tz);
+		efree(dummy_t);
+		timelib_free(dummy_t->tz_abbr);
```

Comment:
`dummy_t->tz_abbr` can be 0 here with `timezone_open("GMT+00:00x")`

Reply:
That might not matter, assuming that `timelib_free(NULL)` is a nop.

Reply:
Yeah, I'm leaving this.